java/vector/src/main/java/org/apache/arrow/vector/types/pojo/Field.java

```diff
@@ -37,6 +40,7 @@
   private final String name;
   private final boolean nullable;
   private final ArrowType type;
+  private final Long dictionary;
```

Comment:
The dictionary id is what goes in the IPC metadata, but how will this work in a strictly in-memory setting? In normal use, when you create a DictionaryVector field, the id will be null here. Then in the file/stream adapter code, you will need to either use the dictionary id here or come up with a new one.

At least in C++, I have planned for the dictionary id to be an encapsulated detail of the stream/file formats, but is otherwise not part of the public API. So I guess what I am asking is whether the dictionary id might have any use in Java outside of files and streams.